pyls/workspace.py

```diff
@@ -159,14 +163,13 @@ def __init__(self, uri, source=None, version=None, local=True, extra_sys_path=No
         self._local = local
         self._source = source
         self._extra_sys_path = extra_sys_path or []
-        self._rope_project = rope
+        self._rope_project_builder = rope_project_builder
 
     def __str__(self):
         return str(self.uri)
 
-    @property
-    def _rope(self):
-        return libutils.path_to_resource(self._rope_project, self.path)
+    def _rope_resource(self, rope_config):
```

Comment:
why is this a `private method` if it is only used outside of the class?

Reply:
Sort of abused the "private" convention to keep the rope APIs hidden. I consider them experimental / unstable and might even be removed in the future.